src/libraries/System.Net.Http/src/System/Net/Http/Headers/WarningHeaderValue.cs

```diff
@@ -203,16 +203,19 @@ internal static int GetWarningLength(string? input, int startIndex, out object?
             return current - startIndex;
         }
 
-        private static bool TryReadAgent(string input, int startIndex, ref int current, [NotNullWhen(true)] out string? agent)
+        private static bool TryReadAgent(string input, ref int current, [NotNullWhen(true)] out string? agent)
         {
-            int agentLength = HttpRuleParser.GetHostLength(input, startIndex, true, out agent!);
+            agent = null;
 
+            int agentLength = HttpRuleParser.GetHostLength(input, current, true);
             if (agentLength == 0)
             {
                 return false;
             }
 
+            agent = input.Substring(current, agentLength);
```

Comment:
Is it worth storing current into a temporary and then moving this Substring down to just before we `return true` so that we're not doing this Substring for the case below where we return false?

Reply:
I agree that that would be better, but (a) this is super rare to do -- parsing a Warning header, and (b) there are a ton of places in various header parsing code where we do stuff like this already. So fixing this one doesn't seem worth it.